pilot/pkg/serviceregistry/kube/controller/multicluster.go

```diff
@@ -314,6 +319,31 @@ func (m *Multicluster) initializeCluster(cluster *multicluster.Cluster, kubeRegi
 	return nil
 }
 
+// checkCanLead returns true if the caller has sufficient authority to become elected leader.
+func (m *Multicluster) checkCanLead(client kubelib.Client) bool {
+	if !features.ExternalIstiod {
+		return false
+	}
+	s := &authorizationapi.SelfSubjectAccessReview{
+		Spec: authorizationapi.SelfSubjectAccessReviewSpec{
+			ResourceAttributes: &authorizationapi.ResourceAttributes{
+				Namespace: m.opts.SystemNamespace,
+				Verb:      "create",
+				Group:     "",
+				Resource:  "configmaps",
+			},
+		},
+	}
+	log.Infof("check resource:\n%v", s)
+	response, err := client.Kube().AuthorizationV1().SelfSubjectAccessReviews().Create(context.TODO(), s, metav1.CreateOptions{})
+	log.Infof("check response:\n%v", response)
+	if err != nil {
+		log.Errorf("could not determine leadership permission: %v", err)
+		return true
+	}
+	return response.Status.Allowed
```

Comment:
This always retruns false???

@howardjohn do you see anything wrong with the above SelfSubjectAccessReview call?

Reply:
Not really, seems correct to me. I am, however, fundamentally opposed with doing this type of check rather than having Istiod be aware of what role it has.

Reply:
Yeah, OK, I'm not crazy about this approach either.